### DISCUSSION TITLE
Data Importer password protection

### DISCUSSION THREAD

**Discussion by M1scer** [firefly-iii/firefly-iii#7839] [Firefly III] [answered]
Discussion about the closed issue #7838 

According to the [documentation ](https://docs.firefly-iii.org/data-importer/advanced/automation/)i need to setup a access toke to use the "POST" command.

How can i do it in this case?

**Comment by JC5** [comment]
Sorry, I should have turned it into a discussion. Use `-e WEB_SERVER=false` so the data importer does not start the UI.

**Reply by M1scer** [reply]
Good idea, I will look at and see if that fits for me. 
Spontaneously, however, I can think of two disadvantages:
- If I want to do auto import and manual import (for different bank accounts) I always have to start and stop as UI.
- This is more of a guess, since I haven't been using Firefly that long: the account approval for Nordigen is limited to 90 days, I assume I have to do a manual import every 90 days to reconfirm the approval via TAN. This would also require the UI to be restarted and closed.

**Reply by M1scer** [reply]
I assume that a basic auth via a reverse proxy like Traefik causes problems with the connection between Importer and Firefly?

**Reply by M1scer** [reply, marked as answer]
Just tried it and seems to work with basic auth activated.
At least if the connection between firefly and the importer is already established.

For me this is the most suitable solution so far.

Thanks.

**Comment by M1scer** [comment]
Unfortunately, I have encountered a new problem caused by the Basic auth.

I use cURL POST command to automate my import. Due to the basic auth I have to specify my login additionally. Unfortunately, I have not been able to do it successfully so far. I always get the error code `401 unauthorized`

I have already looked in the cURL documentation and on the internet and I think that I am using the command correctly.
The command i tried:
`curl -v -u 'user:password' --location --request POST 'https://importer.domain.de/autoupload?secret=xxxxx' --header 'Accept: application/json' --header 'Authorization: Bearer ey....' --form 'json=@/home/import.json'`

Is there anything to note from the importer to get the basic auth to run?

**Reply by M1scer** [reply]
I just realized that the "Brearer authorization" header is not needed. If i modify my command like this:
`curl -v -u 'user:password' --location --request POST 'https://importer.domain.de/autoupload?secret=xxxxx' --header 'Accept: application/json' --form 'json=@/home/import.json'`

it works fine.

Edit:
I tried to run the POST command in the import container itself this is working as well.
So you dont need the -u option, just replace the URL with localhost:port
`curl -v --location --request POST 'https://localhost:8080/autoupload?secret=xxxxx' --header 'Accept: application/json' --form 'json=@/home/import.json'`
